doc/topics/development/contributing.rst

```diff
@@ -4,481 +4,604 @@
 
-    .. code-block:: bash
+::
 
-        git fetch upstream
+   zmq.core.error.ZMQError: ipc path "/path/to/your/virtualenv/var/run/salt/minion/minion_event_7824dcbcfd7a8f6755939af70b96249f_pub.ipc" is longer than 107 characters (sizeof(sockaddr_un.sun_path)).
 
-#.  Update your copy of the ``master`` branch.
+This means that the path to the socket the minion is using is too long.
+This is a system limitation, so the only workaround is to reduce the
+length of this path. This can be done in a couple different ways:
 
-    .. code-block:: bash
+1. Create your virtualenv in a path that is short enough.
+2. Edit the :conf_minion:``sock_dir`` minion config variable and reduce
+   its length. Remember that this path is relative to the value you set
+   in :conf_minion:``root_dir``.
 
-        git checkout master
-        git merge --ff-only upstream/master
+NOTE: The socket path is limited to 107 characters on Solaris and Linux,
+and 103 characters on BSD-based systems.
 
-    If Git complains that a fast-forward merge is not possible, you have local
-    commits.
+No permissions to access …
+~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-    * Run ``git pull --rebase origin master`` to rebase your changes on top of
-      the upstream changes.
-    * Or, run ``git branch <branch-name>`` to create a new branch with your
-      commits. You will then need to reset your ``master`` branch before
-      updating it with the changes from upstream.
+If you forget to pass your config path to any of the ``salt*`` commands,
+you might see
 
-    If Git complains that local files will be overwritten, you have changes to
-    files in your working directory. Run ``git status`` to see the files in
-    question.
+::
 
-#.  Update your fork.
+   No permissions to access "/var/log/salt/master", are you running as the
+   correct user?
 
-    .. code-block:: bash
+Just pass ``-c local/etc/salt`` (or whatever you named it)
 
-        git push origin master
+File descriptor limit
+~~~~~~~~~~~~~~~~~~~~~
 
-#.  Repeat the previous two steps for any other branches you work with, such as
-    the current release branch.
+You might need to raise your file descriptor limit. You can check it
+with:
 
-Posting patches to the mailing list
------------------------------------
+::
 
-Patches will also be accepted by email. Format patches using `git
-format-patch`_ and send them to the `salt-users`_ mailing list. The contributor
-will then get credit for the patch, and the Salt community will have an archive
-of the patch and a place for discussion.
+   ulimit -n
 
-Issue and Pull Request Labeling System
---------------------------------------
+If the value is less than 2047, you should increase it with:
 
-SaltStack uses several labeling schemes to help facilitate code contributions
-and bug resolution. See the :ref:`Labels and Milestones
-<labels-and-milestones>` documentation for more information.
+::
 
-Mentionbot
-----------
+   ulimit -n 2047
+   # For c-shell:
+   limit descriptors 2047
```

Comment:
I did an audit of other open PRs, just in case there may need to be additional edits here, and I noticed this in #56391:

https://github.com/saltstack/salt/pull/56391/files?file-filters%5B%5D=.rst#diff-d4ef981b7725173c30d3e321db8d335dbf3419243e13699514cd1efad2d9f094R244-R247

The `ulimit` value is being adjusted to `3072`. I'm not sure whether this is required or not, as I haven't gone this deep into the development where I was needing to adjust my `ulimit -n` value away from my dev system default of `1024` to `2047` or greater.

Whatever we decide on that (to update the value or not), we could actually close out #56391 because the rest of that PR is specific to Python 2, which is no longer supported by current versions of Salt. I've closed out #58128 as this PR will also supersede the need for it.